Ducky-Chickens/bangazonCLI
# Product Popularity

# Description
Add view product popularity to show top 3 products and their revenue.

Also modified press enter to continue, and added addOrder and addOrderProducts, which Hunter is working on currently.  I needed them to test my feature.

## Related Ticket(s)
Finish Issue #11  (User should be able to see overall product popularity)

## Expected Behavior
When you select see product popularity, after choosing a customer, you will see something similar to the below table.
     ```
       Product             Orders     Purchasers     Revenue
       *******************************************************
       Intelligent Frozen  2          1              $50
       Chips
       Practical Fresh     2          2              $50
       Fish
       Handcrafted Fresh   1          1              $25
       Bacon
       *******************************************************
       Total:              5          4              $125
     ```

## Steps to Test Solution
1. Run `npm install`
1. Run `npm run db:reset`
1. Run `npm i -g`
1. Run `bangazon`
1. Select option `2` to select a customer. (Customer with id 6 is used in the below example).
     1. The available option should be:
          ```
            1. Create a customer account
            2. Choose active customer
            3. Create a payment option
            4. Add product to inventory
            5. Complete an order
            6. See product popularity
            7. View stale products
            . Leave Bangazon!
          ```
1. Select option `6` to view the most popular products.
     ```
       Product             Orders     Purchasers     Revenue
       *******************************************************
       Practical Cotton    1          1              $25
       Chair
       Practical Steel     1          1              $25
       Towels
       *******************************************************
       Total:              2          2              $50
     ```
1. Press enter to continue back to the main menu.

1. Run `npm test`
1. All test to do with product popularity should pass.

## Testing


## Documentation
